datalad/local/remove.py

```diff
@@ -72,6 +72,23 @@ class Remove(Interface):
     subdatasets are located under a to-be-removed path, they will be
     uninstalled entirely, and all their content dropped. If any subdataset
     can not be uninstalled safely, the remove operation will fail and halt.
+
+    .. versionchanged:: 0.16
+       More in-depth and comprehensive safety-checks are now performed by
+       default.
+       The ``if_dirty||--if-dirty`` argument is ignored, will be removed in
+       a future release, and can be removed for a safe-by-default behavior. For
+       other cases consider the ``reckless||--reckless`` argument.
+       The ``save||--save`` argument is ignored and will be removed in a future
+       release, a dataset modification is now always saved. Consider save's
+       ``amend||--amend`` argument for post-remove fix-ups.
+       The ``recursive||--recursive`` argument is ignored, and will be removed
+       in a future release. Removal operations are always recursive, and the
+       parameter can be stripped from calls for a safe-by-default behavior.
```

Comment:
FWIW, and I know that I am late to the discussion, but IMHO `-r` should have remained there for a user to make a conscious decision to remove including underlying subdatasets, similarly to how `rm` would not remove a directory without `-r`.